### DISCUSSION TITLE
*Some* assets not being prefixed with `basePath` when production build deployed to nginx-proxied env

### DISCUSSION THREAD

**Discussion by willmeierart** [vercel/next.js#60787] [App Router]
I can't open an issue on this because the reproduction steps are so specific, but I am fairly certain this is a bug.

Basically, I have an app where the file structure in the app router looks like:

<img width="289" alt="Screenshot 2024-01-17 at 12 28 22 PM" src="https://github.com/vercel/next.js/assets/26130782/14481f14-2cc8-45d3-85cb-591ee05758ac">

, where the `@detail` parallel route slot is a sidebar.

When I build this project locally, everything works completely fine. However, when I deploy the app to a production environment sitting behind an nginx proxy with a `basePath` in the config, the detail route breaks. After a lot of investigating I have been able to determine, upon refreshing the page on this detail route, that some js assets are 404'ing. Inspecting the html of the page response, I can see that the script tags for these assets do not have the `basePath` prefixed.

<img width="1243" alt="Screenshot 2024-01-17 at 12 38 26 PM" src="https://github.com/vercel/next.js/assets/26130782/c47f3f85-c429-4af6-87ae-8c789413566f">

The two aspects of this issue that are so confusing in conflict with one another are: a) since I can see the dynamically generated html has inconsistent filepath prefixing, this HAS to be an issue with next, right? and yet b) why is it that when I build this same project locally, the assets all have the `basePath`? I've built many versions of this code and the results are consistent, and am running the same version of node etc. in the deployed version.

**Comment by luccasmtxr** [comment]
i'm pretty much facing the same issue while deploying a docker image on k8s.
Everything runs fine locally but the static assets break on production and are missing the prefix too

**Reply by willmeierart** [reply]
we are using docker as well when deploying, not kubernetes though.

**Reply by FabrizioArmango** [reply]
Hey, I'm currently facing the same issue.

**Comment by jelsinor** [comment]
Same here

**Comment by thesarangsaraf** [comment]
I am also facing same problem. My next js version is 13.5.6 with node 16. When i run app locally with assetprefix then index is loaded correctly but after deployment (docker) then the prefix is going away.

One weired observation is -
if i add random chars in the url for example host/random then assets prefix is loaded in the html but then it becomes host/random/prefix/file.js while loading assets which results into 404 again.

**Comment by andrewasd** [comment]
hello, is there a fix for this problem, im facing the same issue,
when i run` npm run dev` is working fine,

but if I do `npm run build`, and then `npm run start` is loosing the asset prefix.

**Reply by thesarangsaraf** [reply]
Hello, if you are using ^ mark in your package.json please remove that for nextjs package and rebuild app again. it is issue in nextjs version not in the code. this way it will use specific version instead of higher.

**Comment by farizaan** [comment]
I have the same problem. nextjs version - 15.1.3
in production only some files prefixed by basePath, but locally everything is ok

**Comment by amirzandi** [comment]
Has anyone find a solution for this bug so far?

**Comment by Crenshinibon** [comment]
Same issue. Here Next 15; Node 22

**Comment by akantic** [comment]
Same problem and it's driving me mad. Anyone figured it out?

**Reply by akantic** [reply]
I managed to fix it. It was exhausting to debug it, since there were no indications that the issue is in the build process, but it was a trivial mistake in my case.

I forgot to copy the next config into my production container during the build in Dockerfile. Hopefully it will someone some time, good luck. 🙌

**Reply by VladimirJarabica** [reply]
Thanks! You saved me a lot of headache. Here is a chocolate bar for you 🍫

**Comment by wissem-ensi** [comment]
Same problem as well

**Comment by levinqdl** [comment]
> I managed to fix it. It was exhausting to debug it, since there were no indications that the issue is in the build process, but it was a trivial mistake in my case.
> 
> I forgot to copy the next config into my production container during the build in Dockerfile. Hopefully it will someone some time, good luck. 🙌

This fixes my problem.

**Comment by b-rajkumar** [comment]
A common issue in Next.js applications is that the app works correctly during development using yarn dev or npm run dev, but fails in the production environment after running next build and next start.

This typically happens when the basePath is configured using environment variables or some dynamic method. If these environment variables are not set at build time, Next.js will treat the basePath as undefined or an empty string (""). As a result, the app is built with an incorrect or missing base path, causing routing and asset resolution issues in production.

**Solution:**
To avoid this, ensure that any environment variables used for setting basePath (or similar configuration options) are defined and available during the next build process. This ensures that Next.js correctly applies the basePath and generates the appropriate routes and assets paths for production.

**Comment by sweetcard** [comment]
https://github.com/vercel/next.js/discussions/34599#discussioncomment-10832885 
This Link may be help.
